app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1336,27 +1351,54 @@ public void onDialogDismissed(int dialogId) {
     public static class DatePickerFragment extends DialogFragment
             implements DatePickerDialog.OnDateSetListener {
 
-        final Context context;
-        final EditText textFieldEdit;
-        @Nullable
-        final Date minDate;
-        @Nullable
-        final Date maxDate;
+        private static final String TEXT_FIELD_TO_EDIT_ARGUMENT_KEY = "text_field_to_edit";
+        private static final String CURRENT_DATE_ARGUMENT_KEY = "current_date";
+        private static final String MIN_DATE_ARGUMENT_KEY = "min_date";
+        private static final String MAX_DATE_ARGUMENT_KEY = "max_date";
+        private static final String PICK_DATE_REQUEST_KEY = "pick_date_request_key";
+        private static final String NEWLY_PICKED_DATE_ARGUMENT_KEY = "newly_picked_date";
+
+        public static DatePickerFragment newInstance(@NonNull LoyaltyCardField textField, @Nullable Date currentDate, @Nullable Date minDate, @Nullable Date maxDate) {
+            Bundle args = new Bundle();
+            args.putSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY, textField);
+            args.putSerializable(CURRENT_DATE_ARGUMENT_KEY, currentDate);
+            args.putSerializable(MIN_DATE_ARGUMENT_KEY, minDate);
+            args.putSerializable(MAX_DATE_ARGUMENT_KEY, maxDate);
+            DatePickerFragment fragment = new DatePickerFragment();
+            fragment.setArguments(args);
+            return fragment;
+        }
 
-        DatePickerFragment(Context context, EditText textFieldEdit, @Nullable Date minDate, @Nullable Date maxDate) {
-            this.context = context;
-            this.textFieldEdit = textFieldEdit;
-            this.minDate = minDate;
-            this.maxDate = maxDate;
+        public interface OnDatePickListener {
+            void onDatePicked(@NonNull LoyaltyCardField textFieldToEdit, @NonNull Date newDate);
         }
 
+        public static void registerDatePickListener(@NonNull AppCompatActivity activity, @NonNull OnDatePickListener listener) {
+            activity.getSupportFragmentManager().setFragmentResultListener(
+                    PICK_DATE_REQUEST_KEY,
+                    activity,
+                    (requestKey, result) -> listener.onDatePicked(
+                            (LoyaltyCardField) Objects.requireNonNull(result.getSerializable(TEXT_FIELD_TO_EDIT_ARGUMENT_KEY)),
+                            (Date) Objects.requireNonNull(result.getSerializable(NEWLY_PICKED_DATE_ARGUMENT_KEY))));
+        }
+
+        LoyaltyCardField textFieldEdit;
+        @Nullable
+        Date minDate;
+        @Nullable
+        Date maxDate;
```

Comment:
Nitpick: I would've personally preferred these statements below the `private static final String` and above the `newInstance`. So, at line 1360 at the time of writing :)